table/block_based_table_reader.h

```diff
@@ -603,7 +603,23 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
         is_index_(is_index),
         key_includes_seq_(key_includes_seq),
         index_key_is_full_(index_key_is_full),
-        for_compaction_(for_compaction) {}
+        for_compaction_(for_compaction) {
+
+    if (!for_compaction) {
```

Comment:
Not for this PR, but eventually we should do it for compaction too.

Reply:
yup, agreed. I didn't prioritize it in this PR.